src/include/mutex_inline.h

```diff
@@ -266,6 +292,40 @@ __wt_spin_unlock(WT_SESSION_IMPL *session, WT_SPINLOCK *t)
 
 #endif
 
+/*
+ * __wt_spin_owns --
+ *     Check whether the session owns the spinlock.
+ */
+static inline bool
+__wt_spin_owns(WT_SESSION_IMPL *session, WT_SPINLOCK *t)
+{
+    return (t->session_id == WT_SPIN_SESSION_ID_SAFE(session));
+}
+
+/*
+ * __wt_spin_unlock_if_owns --
+ *     Unlock the spinlock only if it is acquired by the specified session.
+ */
+static inline void
+__wt_spin_unlock_if_owns(WT_SESSION_IMPL *session, WT_SPINLOCK *t)
+{
+    if (__wt_spin_owns(session, t))
+        __wt_spin_unlock(session, t);
+}
+
+/*
+ * WT_ASSERT_SPINLOCK --
+ *      Assert that the session owns the spinlock.
+ */
+#define WT_ASSERT_SPINLOCK(session, t) WT_ASSERT((session), __wt_spin_owns((session), (t)));
```

Comment:
Rename to WT_ASSERT_SPINLOCK_OWNED or WT_ASSERT_HAVE_SPINLOCK? Or WT_REQUIRE_SPINLOCK?